src/concat/requireify/pack-files.js

```diff
@@ -0,0 +1,121 @@
+var _ = require('underscore');
+var browserPack = require('browser-pack');
+var concat = require('concat-stream');
+var FileType = require('../../file').type;
+var Promise = require('bluebird');
+var sourceMap = require('../../file/source-map');
+var toStream = require('string-to-stream');
+
+var exportsRegex = /(module.exports\s*=)|(exports\s*=)|(exports\.[A-Za-z0-9_]+\s*=)/mg;
+
+/**
+ * @param {object} options
+ * @param {object} options.allFiles
+ * @param {object} options.deps
+ * @param {object} options.indices
+ * @param {boolean} options.sourceMap
+ * @param {object} options.exports
+ * @returns {Promise}
+ */
+function pack(options) {
+
+    return new Promise(function(resolve, reject) {
+
+        var packInput = getPackInput(options);
+
+        toStream(JSON.stringify(packInput)).pipe(browserPack())
+            .pipe(concat(function(stream) {
+
+                var result = sourceMap.extract(stream.toString());
+
+                result.code = exposeExports(result.code, options.exports, options.indices);
+
+                if (options.sourceMap) {
+                    resolve(result);
+                } else {
+                    resolve({
+                        code: result.code
+                    });
+                }
+
+            }))
+            .on('error', function(err) {
+                reject(err);
+            });
+
+    });
+
+}
+
+function getPackInput(options) {
+
+    var packInput = [];
+
+    _.each(options.allFiles, function(file, filePath) {
+
+        var fileDeps = {},
+            code;
+
+        _.each(options.deps[filePath], function(dep, reference) {
+            if (dep.isPath) {
+                fileDeps[reference] = options.indices[dep.name];
+            } else {
+                fileDeps[reference] = reference;
+            }
+        });
+
+        if (options.sourceMap && file.map) {
+            code = file.code + '\n' + sourceMap.getComment(file.map, FileType.JS);
+        } else {
+            code = file.code;
+        }
+
+        packInput.push({

```

Comment:
The `browser-pack` npm module expects an array of files as input. Each file looks like this.

- `sourceFile` The path of the file.
- `id` The index of the file.
- `source` The source code of the file (already compiled at this point if it's JSX/ES6/Mustache.
- `deps` A mapping from the `require` calls in the file to the index of the files they refer to.
- `entry` Whether or not to immediately invoke the code when the bundle is first loaded in the browser. By default, code is now lazily invoked, so it won't ever get run until somebody `requires` it. But we're going to have cases where a bundle might not be 100% require-ified code, and we'll want that legacy code to get invoked immediately. An example would be `sg.patient-beta.js.bundle`, where the timesgrid code is going to get `require`-ified, but the other `ModalView`, `StickyView` etc code will not. We want that old code to get run right away so people can say `ZD.module('sticky')` and it'll still work as before. `isEntry` scans the source code of the file, and if it has no exports, we are going to invoke it immediately.